sdk/dotnet/Pulumi/Deployment/DeploymentInstance.cs

```diff
@@ -0,0 +1,54 @@
+using System.Threading.Tasks;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// Metadata of the deployment that is currently running. Accessible via <see cref="Deployment.Instance"/>.
+    /// </summary>
+    public sealed class DeploymentInstance : IDeployment
```

Comment:
Any reason why we can't use the existing `Deployment` class instead of introducing a new public type?

Reply:
Great question!
First, [this line](https://github.com/pulumi/pulumi/blob/a95a4d11954fb28b80f585cd399e570a1a6f80ea/sdk/dotnet/Pulumi.Tests/PulumiTest.cs#L23) in `PulumiTest` would be hard to re-implement.
Second, `Deployment` is currently effectively a static class: it implemented `IDeployment` only explicitly, and not directly. So, I'd have to change the `Deployment` class to add those methods to the class level.
I thought it would make sense to have this separation of the deployment "engine" and its state.
Would love to hear your opinion!

Reply:
Ah, bummer it wasn't easier to tease these apart, but makes sense!